jbpm-wb-common/jbpm-wb-common-client/src/main/java/org/jbpm/workbench/common/client/list/AdvancedSearchFiltersViewImpl.java

```diff
@@ -115,11 +125,77 @@ public void addNumericFilter(final String label,
     }
 
     @Override
+    public void addDateRangeFilter(final String label,
+                                   final Consumer<DateRange> addCallback,
+                                   final Consumer<DateRange> removeCallback) {
+        final UTCDateBox fromDate = GWT.create(UTCDateBox.class);
+        final UTCDateBox toDate = GWT.create(UTCDateBox.class);
+        fromDate.addValueChangeHandler(e -> onDateValueChange(label,
+                                                              fromDate,
+                                                              toDate,
+                                                              addCallback,
+                                                              removeCallback));
+        toDate.addValueChangeHandler(e -> onDateValueChange(label,
+                                                            fromDate,
+                                                            toDate,
+                                                            addCallback,
+                                                            removeCallback));
+
+        final Div div = (Div) getDocument().createElement("div");
+        div.setAttribute("data-filter",
+                         label);
+        div.getClassList().add("input-group");
+        div.getClassList().add("filter-control");
+        div.getClassList().add("hidden");
+        appendWidgetToElement(div,
+                              fromDate);
+        final Div divTo = (Div) getDocument().createElement("div");
+        divTo.getClassList().add("input-group-addon");
+        divTo.setTextContent("to");
+        div.appendChild(divTo);
+        appendWidgetToElement(div,
+                              toDate);
+        filtersInput.appendChild(div);
+        if (filterText.getTextContent().isEmpty()) {
+            setCurrentFilter(label);
+        }
+        createFilterOption(label);
+    }
+
+    protected void onDateValueChange(final String label,
+                                   final UTCDateBox fromDate,
+                                   final UTCDateBox toDate,
+                                   final Consumer<DateRange> addCallback,
+                                   final Consumer<DateRange> removeCallback) {
+        if (toDate.getValue() == null || fromDate.getValue() == null) {
+            return;
+        }
+        final Date toDateValue = UTCDateBox.utc2date(toDate.getValue());
+
+        final Date fromDateValue = UTCDateBox.utc2date(fromDate.getValue());
+        final DateRange dateRange = new DateRange(fromDateValue,
+                                                  toDateValue);
+        addActiveFilter(label,
+                        translationService.format(Constants.FROM)
```

Comment:
@rbarriuso the general guideline is to use Errai api as much as possible to minimize the refactor for when we move to GWT 3. Not entirely sure if the translation/messages will be affected, but widgets in general are expected to be removed.